### DISCUSSION TITLE
Performance question Layout Component

### DISCUSSION THREAD

**Discussion by Mooskopf** [vercel/next.js#29014] [Help] [answered]
Hi,
I have a simple performance question. When using Navbar and Footer I am currently importing them in all pages instead of using the Layout component and nesting my App with it. Does this have an effect on the performance of the app? Like does it load slower? I don't think it has but I want to be sure. 
Thanks in advance

**Comment by wilsonneto-swe** [comment, marked as answer]
Hi @Mooskopf , great question!

The two approaches will result in the same... But using the getLayout approach is a way to standardize the way we put the layout on pages, without this you will need to remember and use it the same way in all the pages, and in a big team this would be a problem, and the other thing is the shared state, if you use the layout in the same way on all your pages you already have a shared state between pages, but a simple difference in a page approach will break this shared state... I use both in some projects, and I prefer to standardize the layout flow.
